content/asp.net/getting-started/logging.md

```diff
@@ -1,20 +1,103 @@
+
+- Trace (disabled by default)
+- Debug
+- Information
+- Warning
+- Error
+- Critical
+
+[Learn more about log levels](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging#log-level).
+
+You can filter out messages of low importance by setting a minimum log level when you configure logging:
+
+```c#
+    // only display messages with log level of Information or higher
+    loggerFactory.AddConsole(LogLevel.Information);
+```
+
+Optionally, you can set different log levels based on where the message originates. A common approach is to specify logging options in a configuration file, such as *appsettings.json*, and pass the configuration section into the *AddConsole* method. A sample *appsettings.json* file:
+
+```javascript
+    {
+    "Logging": {
+        "IncludeScopes": false,
+        "LogLevel": {
+        "Default": "Debug",
+        "System": "Information",
+        "Microsoft": "Information"
+        }
+    }
+    }
+```
+
+The above settings set the default log level to display at `Debug` (or above), but set a higher threshold of `Information` for log messages originating from `System` or `Microsoft` sources. By adjusting these settings, you can ensure your own app's messages are not drowned out by built-in logging from the framework.
+
+You learned how to add configuration files in the [Configuration](configuration.md) lesson. You can add more than one file from which to read configuration information, so that the app's quotes come from one file while the logging settings come from another:
+
+```c#
+    public Startup(IHostingEnvironment env)
+    {
+        var builder = new ConfigurationBuilder()
+            .SetBasePath(env.ContentRootPath)
+            .AddJsonFile("quotes.json", optional: false, reloadOnChange: true)
+            .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true);
+
+        Configuration = builder.Build();
+    }
+```
+
+To specify the configuration section when configuring logging, you would use the following code:
+
+```c#
+    loggerFactory.AddConsole(Configuration.GetSection("Logging"));
+```
+
+## Logging Information
+
+Now that you've configured how your app will send logging messages to the console, it's time to see how to actually log information from your app. First, you'll want to use an `ILoggerFactory` to create an instance of a `Logger`. You typically create a logger for the class that is performing the logging, using the factory's `CreateLogger<T>` method. For instance, in *Startup.cs*, in the `Configure` method, you could add the following code to log a message when the method completed.
+
+```c#
+    var logger = loggerFactory.CreateLogger<Startup>();
+    
+    logger.LogInformation("Configure complete.");
+```
+
+Alternately, you can specify the logger source as a string parameter:
+
+```c#
+    var logger = loggerFactory.CreateLogger("Sourcename");
+```
```

Comment:
This is way early in the TOC. It's before MVC, and it's even before DI. Otherwise I would totally agree, but I think we can demonstrate that later (and maybe on an edit pass link forward to where we might do it).